omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/PatientAllergyController2_0Test.java

```diff
@@ -0,0 +1,459 @@
+	}	
+	
+	@Before
+	public void init() throws Exception {
+		executeDataSet(RestTestConstants2_0.ALLERGY_TEST_DATA_XML);
+		executeDataSet(RestTestConstants2_0.OTHER_NON_CODED_CONCEPT_TEST_DATA_XML);
+		Allergen.setOtherNonCodedConceptUuid(RestTestConstants2_0.ALLERGY_OTHER_NON_CODED_UUID);
+	}
+	
+	@Override
+	public String getURI() {
+		return "patient/" + RestTestConstants2_0.PATIENT_UUID + "/allergy";
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getUuid()
+	 */
+	@Override
+	public String getUuid() {
+		return RestTestConstants2_0.ALLERGY_UUID; // allergy does not support uuid
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getAllCount()
+	 */
+	@Override
+	public long getAllCount() {
+		return 4;
+	}
+	
+	/**
+	 *  Delete Allergy
+	 */
+	@Test
+	public void shouldDeleteUniqueAllergy() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Assert.assertFalse(allergy.isVoided());
+		
+		handle(newDeleteRequest(getURI() + "/" + getUuid(), 
+				new Parameter("reason", "unit test")));
+		allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Assert.assertTrue(allergy.isVoided());
+		Assert.assertEquals("unit test", allergy.getVoidReason());
+	}
+	
+	/**
+	 *  Delete Allergy incorrectly
+	 *  throw ObjectNotFoundException because allergy does not exist
+	 */
+	@Test(expected = ObjectNotFoundException.class)
+	public void shouldThrowExceptionWhenDeletingNonExistentAllergy() throws Exception
+	{
+		handle(newDeleteRequest(getURI() + "/" + "NonExistentAllergyUUID", 
+				new Parameter("reason", "unit test")));
+	}
+
+	/**
+	 *  Delete all Allergies
+	 */
+	@Test
+	public void shouldDeleteAllAllergies() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());		
+		Allergies allergies = Context.getPatientService().getAllergies(allergy.getPatient());
+		Assert.assertEquals(Allergies.SEE_LIST, allergies.getAllergyStatus());
+		Assert.assertFalse(allergy.isVoided());
+
+		handle(newDeleteRequest(getURI(), 
+				new Parameter("reason", "unit test")));
+		
+		allergy = Context.getPatientService().getAllergyByUuid(getUuid());		
+		allergies = Context.getPatientService().getAllergies(allergy.getPatient());
+		Assert.assertTrue(allergy.isVoided());
+		Assert.assertEquals("unit test", allergy.getVoidReason());
+		Assert.assertEquals(Allergies.UNKNOWN, allergies.getAllergyStatus());
+	}
+
+	/**
+	 * compareJsonStrings
+	 * - json marshalling makes java null = empty string
+	 */
+	public boolean compareJsonString(String left, String right) {
+		if ((left == null && right == "") || (left == "" && right == null)) {
+			return true;
+		}
+		if (OpenmrsUtil.nullSafeEquals(left, right)) {
+			return true;
+		}
+		return false;
+	}
+	
+	/**
+	 *  GetAllergyReaction from allergy based on allergyReaction

```

Comment:
Since this is simply a method's description, can you put spaces between each word?